components/calendar/calendar.js

```diff
@@ -663,87 +707,122 @@ class Calendar extends LocalizeStaticMixin(RtlMixin(LitElement)) {
 			} case keyCodes.PAGEUP: {
 				const diff = getNumberOfDaysToSameWeekPrevMonth(this._shownMonth, this._shownYear);
 
-				this._focusDate.setDate(this._focusDate.getDate() - diff);
+				const possibleFocusDate = new Date(this._focusDate);
+				possibleFocusDate.setDate(possibleFocusDate.getDate() - diff);
 				this._keyboardTriggeredMonthChange = true;
 
 				// handle when current month has more weeks than previous month and page up pressed from last week
-				if (this._focusDate.getMonth() === this._shownMonth) this._focusDate.setDate(this._focusDate.getDate() - 7);
+				if (possibleFocusDate.getMonth() === this._shownMonth) possibleFocusDate.setDate(possibleFocusDate.getDate() - 7);
 
 				this._updateShownMonthDecrease(true, true);
+				await this.updateComplete;
+				this._updateFocusDateDependentOnDisabled(possibleFocusDate);
+				if (this._focusDate) this._focusDateAddFocus();
```

Comment:
for `PAGEUP`: if there is not a focusable focusDate, focus on the previous month button icon (I'm thinking I should call that out more explicitly instead of using `this.focus()` in case for some reason `focus()` functionality changes.)

Reply:
`PAGEDOWN` works similarly, but focus would go to the next month button if no focusable dates